mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1147,14 +1153,75 @@ def _log_metrics_and_artifacts(self):
         """
         Helper method for generating artifacts, logging metrics and artifacts.
         """
-        if self.model_type == _ModelType.CLASSIFIER:
-            if self.is_binomial:
-                self._log_binary_classifier_artifacts()
-            else:
-                self._log_multiclass_classifier_artifacts()
-            self._log_confusion_matrix()
+        if self.model_type in (_ModelType.CLASSIFIER, _ModelType.REGRESSOR):
+            if self.model_type == _ModelType.CLASSIFIER:
+                if self.is_binomial:
+                    self._log_binary_classifier_artifacts()
+                else:
+                    self._log_multiclass_classifier_artifacts()
+                self._log_confusion_matrix()
+            self._log_model_explainability()
         self._log_metrics()
-        self._log_model_explainability()
+
+    def _log_eval_table(self):
+        metric_prefix = self.evaluator_config.get("metric_prefix", "")
+        if self.dataset.has_targets:
+            data = self.dataset.features_data.assign(
+                **{self.dataset.targets_name or "target": self.y, "outputs": self.y_pred}
+            )
+        else:
+            data = self.dataset.features_data.assign(outputs=self.y_pred)
+        mlflow.log_table(data, artifact_file=f"{metric_prefix}{_EVAL_TABLE_FILE_NAME}")
+
+    def _evaluate_question_answering(self):
+        self._log_eval_table()
+        name = _EVAL_TABLE_FILE_NAME.split(".", 1)[0]
+        self.artifacts[name] = JsonEvaluationArtifact(
+            uri=mlflow.get_artifact_uri(_EVAL_TABLE_FILE_NAME)
+        )
+
+        if self.dataset.has_targets:
+            try:
+                import evaluate
+
+                exact_match = evaluate.load("exact_match")
```

Comment:
Is there a difference between this and https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html? If not, I think we should use https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html, since it doesn't require users to install an additional dependency

Reply:
Great idea!